src/app/components/cohort-charts/chart.service.js

```diff
@@ -677,6 +596,31 @@ angular.module('transmartBaseUi').factory('ChartService',
                 }
             };
 
+
+            /**
+             * Emphasize pie chart legends when the corresponding slices are selected
+             * @memberof ChartService
+             * @param chart
+             * @param el
+             */
+            chartService.emphasizeChartLegend = function (chart, el) {
```

Comment:
using $scope.$digest does not seem to delay the 'expect(ChartService.emphasizeChartLegend).toHaveBeenCalled();' clause to be executed, I keep the initial code where the postRender listener is called and the expectancy is checked there. Also, because postRender is called after renderlet, we can guarantee that emphasizeChartLegend has been called when the postRender is invoked.
 Add a line note